File: src/frontend/src/types/todo.ts
import type { Priority } from './priority'
export type { Priority } from './priority'

/**
 * Todoアイテムの型定義
 */
export interface Todo {
  /** ユニークID（バックエンドが採番する整数） */
  id: number
  /** タスクのテキスト内容 */
  text: string
  /** 完了フラグ */
  completed: boolean
  /** 作成日時（ISO 8601形式） */
  createdAt: string
  /** 優先度（オプション） */
  priority?: Priority
  /** 締め切り日（YYYY-MM-DD形式、オプション） */
  dueDate?: string
}

File: src/backend/src/TodoApp.Api/Models/Todo.cs
namespace TodoApp.Api.Models;

/// <summary>
/// Todoエンティティ
/// </summary>
public class Todo
{
    /// <summary>
    /// 一意識別子
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// タスクの内容（最大200文字）
    /// </summary>
    public string Text { get; set; } = string.Empty;

    /// <summary>
    /// 完了状態
    /// </summary>
    public bool Completed { get; set; }

    /// <summary>
    /// 作成日時（UTC）
    /// </summary>
    public DateTime CreatedAt { get; set; }

    /// <summary>
    /// 期限日（省略可）
    /// </summary>
    public DateOnly? DueDate { get; set; }

    /// <summary>
    /// 優先度ID（省略可）
    /// </summary>
    public string? PriorityId { get; set; }

    /// <summary>
    /// 優先度（ナビゲーションプロパティ）
    /// </summary>
    public Priority? Priority { get; set; }

    /// <summary>
    /// ユーザーID（省略可）
    /// </summary>
    public int? UserId { get; set; }

    /// <summary>
    /// ユーザー（ナビゲーションプロパティ）
    /// </summary>
    public User? User { get; set; }
}
